crates/bitwarden/src/platform/fido2/crypto.rs

```diff
@@ -0,0 +1,195 @@
+use coset::{
+    iana::{self, Algorithm, EnumI64},
+    CoseKey, CoseKeyBuilder,
+};
+use p256::{ecdsa::SigningKey, pkcs8::EncodePrivateKey, SecretKey};
+use passkey::types::ctap2::{AttestedCredentialData, Ctap2Error};
+
+use crate::error::{Error, Result};
+
+pub fn cose_key_to_pkcs8(cose_key: &CoseKey) -> Result<Vec<u8>> {
+    // cose_key.
+    let secret_key = private_key_from_cose_key(cose_key).map_err(|error| {
+        log::error!("Failed to extract private key from cose_key: {:?}", error);
+        Error::Internal("Failed to extract private key from cose_key".into())
+    })?;
+
+    let vec = secret_key
+        .to_pkcs8_der()
+        .map_err(|error| {
+            log::error!("Failed to convert P256 private key to PKC8: {:?}", error);
+            Error::Internal("Failed to convert P256 private key to PKC8".into())
+        })?
+        .as_bytes()
+        .to_vec();
+
+    Ok(vec)
+}
+
+pub fn pkcs8_to_cose_key(secret_key: &[u8]) -> Result<CoseKey> {
+    let secret_key = SecretKey::from_slice(secret_key).map_err(|error| {
+        log::error!("Failed to extract private key from secret_key: {:?}", error);
+        Error::Internal("Failed to extract private key from secret_key".into())
+    })?;
+
+    let cose_key_pair = CoseKeyPair::from_secret_key(&secret_key, iana::Algorithm::ES256);
+    Ok(cose_key_pair.private)
+}
+
+/// Copied from `passkey-rs`
+/// Extract a cryptographic secret key from a [`CoseKey`].
+// possible candidate for a `passkey-crypto` crate?
+/*
+MIT License
+
+Copyright (c) 2022 1Password
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+fn cose_key_to_vec(cose_key: CoseKey) -> coset::Result<Vec<u8>> {
+    let mut data = Vec::new();
+    use coset::AsCborValue;
+    coset::cbor::ser::into_writer(&cose_key.to_cbor_value()?, &mut data)?;
+    Ok(data)
+}
```

Comment:
We should just expose the function in passkey-rs for now and revisit when we look into contributing upstream.

Reply:
Done in 1e8c37fc78ce35d9e0f3ea2fdb89af00853c39ec